api/options.c

```diff
@@ -272,7 +272,8 @@ NLOPT_STDCALL nlopt_set_lower_bounds(nlopt_opt opt, const double *lb)
      nlopt_unset_errmsg(opt);
      if (opt && (opt->n == 0 || lb)) {
 	  unsigned int i;
-	  memcpy(opt->lb, lb, sizeof(double) * (opt->n));
+         if (opt->lb)
```

Comment:
It seems a lot clearer to me to check `opt->n > 0`, and more reliable too: it is possible in the future that we might have a non-NULL pointer here even when `n == 0`, and then you would get a warning if `lb == 0`.

Reply:
ok